src/android/android-backend.cpp

```diff
@@ -0,0 +1,73 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+#ifdef RS2_USE_ANDROID_BACKEND
+
+
+#include "android-backend.h"
+#include "android-uvc.h"
+#include "android-hid.h"
+#include "../types.h"
+#include "usb_host/device_watcher.h"
+#include "usb_host/device_enumerator.h"
+#include <chrono>
+#include <cctype> // std::tolower
+
+namespace librealuvc {
+    namespace platform {
+        std::shared_ptr<uvc_device> android_backend::create_uvc_device(uvc_device_info info) const {
+
+            LOG_DEBUG("Creating UVC Device from path: " << info.device_path.c_str());
+            return std::make_shared<retry_controls_work_around>(
+                    std::make_shared<android_uvc_device>(info, shared_from_this()));
+        }
+
+        std::shared_ptr<backend> create_backend()
+        {
+            return std::make_shared<android_backend>();
+        }
+
+        std::vector<uvc_device_info> android_backend::query_uvc_devices() const {
+            // Ultimately we will update devices on USB hotplug events but for the time being
+            // we're satisfied with a persistent cache
+            auto devices = usb_host::enumerate_usb_devices();
+
+            for (auto dev : devices) {
+              usb_host::android_device_watcher::addUsbDevice(dev.first, dev.second);
+            }
+
+            return usb_host::android_device_watcher::query_uvc_devices();
+        }
+
+        std::shared_ptr<usb_device> android_backend::create_usb_device(usb_device_info info) const {
+            throw std::runtime_error("create_usb_device Not supported");
+        }
+
+        std::vector<usb_device_info> android_backend::query_usb_devices() const {
+
+            std::vector<usb_device_info> result;
+            // Not supported
```

Comment:
This method never gets called but I wonder if it would be better to throw a `std::runtime_error` like the method above if it's not supported in order to be consistent.

Reply:
Fixed in 3b51eef86, along with the other unsupported function that doesn't throw. I think `std::runtime_error` is not the right exception for this, but that's a discussion for a different time.